pytorch_lightning/utilities/parsing.py

```diff
@@ -160,18 +173,22 @@ def collect_init_args(frame, path_args: list, inside: bool = False) -> list:
           most specific class in the hierarchy.
     """
     _, _, _, local_vars = inspect.getargvalues(frame)
-    if '__class__' in local_vars:
-        local_args = get_init_args(frame)
-        # recursive update
-        path_args.append(local_args)
-        return collect_init_args(frame.f_back, path_args, inside=True)
-    elif not inside:
-        return collect_init_args(frame.f_back, path_args, inside)
+    # frame.f_back must be of a type types.FrameType for get_init_args/collect_init_args due to mypy
+    if isinstance(frame.f_back, types.FrameType):
```

Comment:
can you invert the if and return path_args first, then unindent the block below?

Reply:
Thanks, good point! I changed that :)

Reply:
wonderful!